### PR TITLE
fix: `Image` not defined on node

### PR DESCRIPTION
**Pull request by JounQin**
fix https://github.com/liximomo/lazy-compile-webpack-plugin/issues/5#issuecomment-549103432

### CODE REVIEW

**Comment by liximomo** [comment]
Any user case for extra ignore rules?

**Comment by JounQin** [comment]
@liximomo In Angular app, it requires plain string `html` and `style` contents.

**Comment by JounQin** [comment]
@liximomo If you don't mind, you can checkout https://github.com/rx-ts/pkgr/blob/master/packages/webpack/package.json#L43 and change it to published latest version `@0.1.10` and then run `yarn && yarn dev:angular` at project root dir.

It's a bit big repo, sorry for it.

File: lib/plugin.js [hunk 1]
```diff
@@ -37,7 +37,11 @@ class LazyCompilePlugin {
     this.options = Object.assign(
       {
         refreshAfterCompile: false,
-        ignores: [/html-webpack-plugin/],
+        ignores: [
```
Comment (liximomo) [inline lib/plugin.js]: These rules should not be included in the plugin. We should pass it in options:

```js
new LazyCompileWebpackPlugin({
	ignores: [
      /\bhtml-webpack-plugin\b/,
      /\b(html|raw|to-string)-loader\b/,
      /\bexports\.toString\b/
    ]
})
```

Comment (JounQin) [reply lib/plugin.js]: @liximomo Just thought about that either, is `/\bhtml-webpack-plugin\b/` should be included by default always?

Comment (JounQin) [reply lib/plugin.js]: And `raw-loader` should be included by default too in my opinion which emits plain output like `module.exports = ...`

Comment (liximomo) [reply lib/plugin.js]: We need to ignore a loader only if it is used with an entry file. (always true for `html-webpack-plugin`).

Comment (JounQin) [reply lib/plugin.js]: @liximomo Please review again to check whether the description for `ignore` option is correct and suitable.

File: README.md [hunk 2]
```diff
@@ -33,19 +33,37 @@ module.exports = {
...
 ## Options
 
-|                    Name                     |            Type             | Default | Description                                        |
-| :-----------------------------------------: | :-------------------------: | :-----: | :------------------------------------------------- |
-|              **[`refreshAfterCompile`](#refreshAfterCompile)**              |    `boolean`    | `false`  | Enable/Disable *page refresh* when compilation is finish                    |
-
+|                       Name                        |         Type          |   Default   | Description                                                  |
+| :-----------------------------------------------: | :-------------------: | :---------: | :----------------------------------------------------------- |
+| **[`refreshAfterCompile`](#refreshAfterCompile)** |       `boolean`       |   `false`   | Enable/Disable _page refresh_ when compilation is finish     |
+|             **[`ignores`](#ignores)**             | `string[] | RegExp[]` | `undefined` | Match webpack entry request to be ignored from lazy compiler |
```
Comment (liximomo) [inline README.md]: Type should be `Function[] | RegExp[]`. 

`Match webpack entry request to be ignored from lazy compiler` > `Request to be ignored from lazy compiler.`

**Comment by JounQin** [comment]
> Type should be `Function[] | RegExp[]`.
> 
> `Match webpack entry request to be ignored from lazy compiler` > `Request to be ignored from lazy compiler.`

Done.

**Comment by JounQin** [comment]
@liximomo Any other idea or can we just merge and replease this?

**Comment by JounQin** [comment]
@liximomo I also found every lazy loading module from `loadingChildren` of Angular Router was considered as an entry module because it refreshed every time for first loading.

**Comment by JounQin** [comment]
@liximomo Any news??

**Comment by liximomo** [comment]
@JounQin Thanks for your contribution.

**Pull request merged**